### DISCUSSION TITLE
Support in 'C' language

### DISCUSSION THREAD

**Discussion by ameerkilari** [chipsalliance/caliptra-sw#1215] [Q&A] [answered]
Hi,
   Is there any plan to support caliptra-sw in 'C' or 'C++' instead in RUST language?

**Comment by jhand2** [comment, marked as answer]
Caliptra firmware is written in Rust and there are no plans to rewrite it in another language. However Caliptra firmware is self-contained inside the Caliptra hardware boundary, so it should not impact integration into any other SoC firmware/software.

libcaliptra (inside this repo) is a C library for writing client code that interacts with Caliptra's registers, which integrators can use to communicate with Caliptra (mailbox, sha accelerators, etc).
